### PR TITLE
[Merged by Bors] - feat(FieldTheory/Galois): Lemmas of galois theory

### PR DESCRIPTION
**Pull request by Thmoas-Guan**
Add lemmas about Galois extensions, especially the special case of the Galois correspondence.
Co-authored-by:  Yongle Hu <mbkybky@gmail.com> Nailin Guan <3571819596@qq.com> Jingting Wang <wangjingting2023@stu.pku.edu.cn>

---

- [x] depends on: #17197

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by chrisflav** [comment]
This PR seems to have overlap with #16978. Also the "Moves: ..." part of the PR description does not seem accurate any more.

**Comment by Thmoas-Guan** [comment]
This is all the lemmas needed for the incoming PRs #16982 and #16993 , but some is still needed in #16978, do I need to remove the overlaping part or work on this PR first?

**Comment by faenuccio** [comment]
> This is all the lemmas needed for the incoming PRs #16982 and #16993 , but some is still needed in #16978, do I need to remove the overlaping part or work on this PR first?

Do you mean that #16978 depends on this PR? If this is the case, please mark it there and add the blocked-by-other-PR label (add it also to the other two PR's that depend on this one).

**Comment by Thmoas-Guan** [comment]
#16978 can be handled seperately, but should I remove the overlaping part? Because some lemmas in this PR is also used in #16979

**Comment by faenuccio** [comment]
> #16978 can be handled seperately, but should I remove the overlaping part? Because some lemmas in this PR is also used in #16979

Well, if nothing in #16978 needs what is in this PR, then yes, you should remove the overlap: once this PR will be merged the results will be in `master`  and there will be no need to re-add them through #16978. Even better, as soon as this one is merged, when you will update master on #16978, this will import these new lemmas there, so there will be no need to add them manually. All in all, a certain def/lemma should appear in a PR only if
1. You are trying to add it to master 
2. You need it for the other results of that PR.

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (Thmoas-Guan) [reply Mathlib/FieldTheory/Galois/Basic.lean:315]: Sorry, what API lemma?

Comment (tb65536) [reply Mathlib/FieldTheory/Galois/Basic.lean]: A lemma that says that if you start with a automorphism of `L`, map it to an automorphism of `fixedField H`, apply it to an element of `fixedField H`, and coerce to `L`, then you get the same answer as if you just immediately coerce to `L` and apply the automorphism of `L`.

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (Thmoas-Guan) [reply Mathlib/FieldTheory/Galois/Basic.lean]: Restated, the proof is simplified.

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (Thmoas-Guan) [reply Mathlib/FieldTheory/Galois/Basic.lean]: Added after `AlgEquiv.restrictNormalHom`, named as `AlgEquiv.restrictNormalHom_apply`

File: Mathlib/FieldTheory/Normal.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Normal.lean:288]: Can you also add a version like `AlgEquiv.restrictNormal_commutes` that works in general and not just for intermediate fields?

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:324]: Can you add the same sort of API lemma for this def as well?

File: Mathlib/FieldTheory/Normal.lean
Comment (Thmoas-Guan) [reply Mathlib/FieldTheory/Normal.lean:288]: I think this lemma already exists, but I prefer to still have one for intermediate field because it is more frequently used.
image
This is only an unfold away from restrictNormalHom applying on an element.

Comment (tb65536) [reply Mathlib/FieldTheory/Normal.lean:288]: Fair enough. Although it does feel a bit weird to have `AlgEquiv.restrictNormal` have a general lemma but `AlgEquiv.restrictNormalHom` to have a lemma just for intermediate fields.

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (Thmoas-Guan) [reply Mathlib/FieldTheory/Galois/Basic.lean:324]: Added, but do I need to add one more about the composition of `QuotientGroup.mk'` and `normalAutEquivQuotient` is equal to `restrictNormalHom`

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:328]: Maybe just use the coercion?
```suggestion
    normalAutEquivQuotient H σ = (restrictNormalHom (fixedField H)) σ := rfl
```

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:335]: Maybe there should be a `simp`-lemma `IntermediateField.mem_fixingSubgroup_iff` to avoid having to unfold the definition of `IntermediateField.fixingSubgroup`.

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:341]:
```suggestion
  simp only [coe_map, AlgHom.coe_coe, Set.mem_image, SetLike.mem_coe, AlgEquiv.smul_def,
    forall_exists_index, and_imp, forall_apply_eq_imp_iff₂, Subtype.forall,
    Subgroup.mem_pointwise_smul_iff_inv_smul_mem, ← symm_apply_eq,
    IntermediateField.fixingSubgroup, mem_fixingSubgroup_iff]
  rfl
```

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:350]:
```suggestion
  rw [← fixingSubgroup_conjugate_of_map, normal_iff_forall_map_eq'.mp inferInstance σ]
```

File: Mathlib/Algebra/Group/Subgroup/Pointwise.lean
Comment (tb65536) [inline Mathlib/Algebra/Group/Subgroup/Pointwise.lean:399]: Isn't `{G : Type*} [Group G]` already defined as a variable in this file?

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:293]: Would `IntermediateField.restrictNormalHom_ker` be a better name, to enable dot-notation?

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:328]: Perhaps `IntermediateField.map_fixingSubgroup`?

File: Mathlib/FieldTheory/IntermediateField/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/IntermediateField/Basic.lean:623]: I don't this is used anywhere, but if you intend to keep it, could you add some basic API like `lift_algEquiv_apply`?

(Also, maybe it should be called `liftAlgEquiv` since it's a def and not a theorem?)

File: Mathlib/FieldTheory/IntermediateField/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/IntermediateField/Basic.lean:617]:
```suggestion
  toFun x := ⟨x.1.1, (mem_lift x.1).mpr x.2⟩
```

File: Mathlib/FieldTheory/IntermediateField/Basic.lean
Comment (Thmoas-Guan) [reply Mathlib/FieldTheory/IntermediateField/Basic.lean:623]: This lemma would be used in the incoming PR #16982 because I got trouble proving the lift of a finite dimensional intermediatefield is also finite dimensional, maybe there are better ways to solve it. But atleast for now I should add `liftAlgEquiv_apply`

File: Mathlib/FieldTheory/Galois/Basic.lean
Comment (tb65536) [inline Mathlib/FieldTheory/Galois/Basic.lean:303]:
```suggestion
lemma IntermediateField.restrictNormalHom_ker (E : IntermediateField K L) [Normal K E] :
```

**Review by tb65536 — APPROVED** [review]

**Comment by tb65536** [comment]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**
